src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs

```diff
@@ -650,5 +652,100 @@ public void MetadataPropertyFunctionBug()
 
             ObjectModelHelpers.AssertItems(new[] { $"{prefix}foo" }, items, expectedMetadata);
         }
+
+        [Fact]
+        public void ItemMetadataShouldBeEvaluatedCaseInsensitively()
+        {
+            var projectContents =
+            @"<Project>
+                <ItemGroup>
+                    <Foo Include=`Foo`>
+                        <SensitiVE>X</SensitiVE>
+                    </Foo>
+                </ItemGroup>
+                <Target Name=`AccessItemMetadata`>
+                    <CreateProperty Value=`%(Foo.FileName)=%(Foo.sensitive)`>  
+                        <Output TaskParameter=`Value` PropertyName=`Qualified` />  
+                    </CreateProperty>          
+                    <CreateProperty Value=`%(Foo.FileName)=%(sensitive)`>  
+                        <Output TaskParameter=`Value` PropertyName=`Unqualified` />  
+                    </CreateProperty>  
+                </Target>
+            </Project>".Cleanup();
+
+            ProjectInstance project = new Project(
+                XmlReader.Create(
+                    new StringReader(projectContents)
+                )
+            ).CreateProjectInstance();
+
+            project.Build();
+
+            Assert.NotNull(project.GetProperty("Qualified"));
+            Assert.NotNull(project.GetProperty("Unqualified"));
```

Comment:
Can you change these asserts to additionally check the content of the properties for good measure?